hashicorp/go-tfe
# Add ExpiredAt to Organization,  Team, User Authentication Tokens

## Description

Adds `ExpiredAt` to Organization, Team, and User tokens and updates the accompanying tests. This is part of a larger change which adds a time-to-live to API authentication tokens.

## Testing plan

1. Verify that you can set the properties by using `go-tfe` within the TFE provider.
2. Verify that all tests still pass.

## Output from tests

#### `team_token_integration_test.go` results:
```
=== RUN   TestTeamTokensCreate
--- PASS: TestTeamTokensCreate (2.31s)
=== RUN   TestTeamTokensCreate/with_valid_options
    --- PASS: TestTeamTokensCreate/with_valid_options (0.21s)
=== RUN   TestTeamTokensCreate/when_a_token_already_exists
    --- PASS: TestTeamTokensCreate/when_a_token_already_exists (0.26s)
=== RUN   TestTeamTokensCreate/without_valid_team_ID
    --- PASS: TestTeamTokensCreate/without_valid_team_ID (0.00s)
PASS

Process finished with the exit code 0

...

=== RUN   TestTeamTokens_CreateWithOptions
--- PASS: TestTeamTokens_CreateWithOptions (2.54s)
=== RUN   TestTeamTokens_CreateWithOptions/with_valid_options
    --- PASS: TestTeamTokens_CreateWithOptions/with_valid_options (0.21s)
=== RUN   TestTeamTokens_CreateWithOptions/when_a_token_already_exists
    --- PASS: TestTeamTokens_CreateWithOptions/when_a_token_already_exists (0.22s)
=== RUN   TestTeamTokens_CreateWithOptions/without_valid_team_ID
    --- PASS: TestTeamTokens_CreateWithOptions/without_valid_team_ID (0.00s)
=== RUN   TestTeamTokens_CreateWithOptions/without_an_expiration_date
    --- PASS: TestTeamTokens_CreateWithOptions/without_an_expiration_date (0.22s)
=== RUN   TestTeamTokens_CreateWithOptions/with_an_expiration_date
    --- PASS: TestTeamTokens_CreateWithOptions/with_an_expiration_date (0.21s)
PASS

Process finished with the exit code 0

...

=== RUN   TestTeamTokensRead
--- PASS: TestTeamTokensRead (3.31s)
=== RUN   TestTeamTokensRead/with_valid_options
    --- PASS: TestTeamTokensRead/with_valid_options (0.65s)
=== RUN   TestTeamTokensRead/with_an_expiration_date_passed_as_a_valid_option
    --- PASS: TestTeamTokensRead/with_an_expiration_date_passed_as_a_valid_option (0.65s)
=== RUN   TestTeamTokensRead/when_a_token_doesn't_exists
    --- PASS: TestTeamTokensRead/when_a_token_doesn't_exists (0.20s)
=== RUN   TestTeamTokensRead/without_valid_organization
    --- PASS: TestTeamTokensRead/without_valid_organization (0.00s)
PASS

Process finished with the exit code 0
```

#### `organization_token_integration_test.go` results:
```
=== RUN   TestOrganizationTokensCreate
--- PASS: TestOrganizationTokensCreate (1.62s)
=== RUN   TestOrganizationTokensCreate/with_valid_options
    --- PASS: TestOrganizationTokensCreate/with_valid_options (0.22s)
=== RUN   TestOrganizationTokensCreate/when_a_token_already_exists
    --- PASS: TestOrganizationTokensCreate/when_a_token_already_exists (0.21s)
=== RUN   TestOrganizationTokensCreate/without_valid_organization
    --- PASS: TestOrganizationTokensCreate/without_valid_organization (0.00s)
PASS

Process finished with the exit code 0

...

=== RUN   TestOrganizationTokens_CreateWithOptions
--- PASS: TestOrganizationTokens_CreateWithOptions (2.15s)
=== RUN   TestOrganizationTokens_CreateWithOptions/with_valid_options
    --- PASS: TestOrganizationTokens_CreateWithOptions/with_valid_options (0.25s)
=== RUN   TestOrganizationTokens_CreateWithOptions/when_a_token_already_exists
    --- PASS: TestOrganizationTokens_CreateWithOptions/when_a_token_already_exists (0.26s)
=== RUN   TestOrganizationTokens_CreateWithOptions/without_valid_organization
    --- PASS: TestOrganizationTokens_CreateWithOptions/without_valid_organization (0.00s)
=== RUN   TestOrganizationTokens_CreateWithOptions/without_an_expiration_date
    --- PASS: TestOrganizationTokens_CreateWithOptions/without_an_expiration_date (0.21s)
=== RUN   TestOrganizationTokens_CreateWithOptions/with_an_expiration_date
    --- PASS: TestOrganizationTokens_CreateWithOptions/with_an_expiration_date (0.21s)
PASS

Process finished with the exit code 0

...

=== RUN   TestOrganizationTokensRead
--- PASS: TestOrganizationTokensRead (2.60s)
=== RUN   TestOrganizationTokensRead/with_valid_options
    --- PASS: TestOrganizationTokensRead/with_valid_options (0.64s)
=== RUN   TestOrganizationTokensRead/with_an_expiration_date_passed_as_a_valid_option
    --- PASS: TestOrganizationTokensRead/with_an_expiration_date_passed_as_a_valid_option (0.63s)
=== RUN   TestOrganizationTokensRead/when_a_token_doesn't_exists
    --- PASS: TestOrganizationTokensRead/when_a_token_doesn't_exists (0.19s)
=== RUN   TestOrganizationTokensRead/without_valid_organization
    --- PASS: TestOrganizationTokensRead/without_valid_organization (0.00s)
PASS

Process finished with the exit code 0
```

#### `user_token_integration_test.go` results:
```
=== RUN   TestUserTokens_Create
--- PASS: TestUserTokens_Create (2.24s)
=== RUN   TestUserTokens_Create/create_token_with_no_description
    --- PASS: TestUserTokens_Create/create_token_with_no_description (0.27s)
=== RUN   TestUserTokens_Create/create_token_with_description
    --- PASS: TestUserTokens_Create/create_token_with_description (0.21s)
=== RUN   TestUserTokens_Create/create_token_without_an_expiration_date
    --- PASS: TestUserTokens_Create/create_token_without_an_expiration_date (0.19s)
=== RUN   TestUserTokens_Create/create_token_with_an_expiration_date
    --- PASS: TestUserTokens_Create/create_token_with_an_expiration_date (0.19s)
PASS

Process finished with the exit code 0
```
